### PR TITLE
WIP: Fix/stammw/clippy fixes

### LINKED ISSUE #59: CI: Make clippy a part of it
Run clippy checks on the pipeline and fix all clippy warnings.

### PR DESCRIPTION
**Pull request by stammw**
fixes #59

### CODE REVIEW

File: examples/client.rs [hunk 1]
```diff
@@ -118,9 +118,9 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         eprintln!("Response: {:?} {}", resp.version(), resp.status());
         eprintln!("Headers: {:#?}", resp.headers());
 
-        while let Some(chunk) = stream.recv_data().await? {
+        while let Some(mut chunk) = stream.recv_data().await? {
             let mut out = tokio::io::stdout();
-            out.write_all(&chunk).await.expect("write_all");
+            out.write_buf(&mut chunk).await.expect("write_all");
```
Comment (FrankSpitulski) [inline examples/client.rs:123]: is this meant to stay write_all, just like here https://github.com/hyperium/h3/pull/67#discussion_r768213749 ?

Comment (stammw) [reply examples/client.rs:123]: It will end up being "write_all_buf()"

**Comment by seanmonstar** [comment]
This probably needs a rebase, seems to include all the changes from #67, but with clippy fixes making it conflict.

**Comment by stammw** [comment]
Ho no! :man_facepalming: Didn't think of that...
There, it's rebased  now.

File: examples/client.rs (hunk 1 above)
Comment (seanmonstar) [reply examples/client.rs:123]: Is this change supposed to be included? It looks like we want it to stay `write_all_buf`.

**Review by seanmonstar — APPROVED** [review]
Looks good, just 1 change I think wasn't meant to be included.

File: examples/client.rs (hunk 1 above)
Comment (stammw) [reply examples/client.rs]: I actually messed up the rebase. Good catch, thanks!

**Pull request merged**